app/src/models/repository.ts

```diff
@@ -77,6 +77,17 @@ export type LinkedWorkTree = WorkingTree & {
   readonly head: string
 }
 
+/** Identical to `Repository`, except it **must** have a `gitHubRepository` */
+export type RepositoryWithGitHubRepository = Repository & {
+  readonly gitHubRepository: GitHubRepository
+}
```

Comment:
One potentially controversial thing I've done is add a `RepositoryWithGitHubRepository` type which guarantees a `Repository` model has a `GitHubRepository` model in it. I did this primarily to make the null checks a little easier to read, since the `PullRequestCoordinator` is only be concerned with repositories that are connected to GitHub somehow, while `AppStore` and `RepositoriesStore` are concerned with all local repositories. This is kind of a band-aid for our domain model but that's a (long) story for another day.